src/components/search/GlobalSearchField.js

```diff
@@ -145,15 +144,30 @@ const getViewAllPrompt = (id, searchTerm, i18NSearch) => {
 };
 
 const SearchOption = React.forwardRef((props, ref) => {
-    const { id, primary, secondary, icon, searchTerm, onClick, href } = props;
+    const {
+        id,
+        primary,
+        secondary,
+        icon,
+        searchTerm,
+        onOptionSelected,
+        href,
+    } = props;
     const classes = useStyles();
     const theme = useTheme();
     return (
         <>
             <span className={classes.optionIcon}>{icon}</span>
             <MuiLink
                 href={href}
-                onClick={onClick}
+                onClick={(event) => {
+                    if (onOptionSelected) {
+                        event.preventDefault();
+                        onOptionSelected();
+                        return false;
+                    }
+                    // else, do default link behavior
+                }}
```

Comment:
I think you cannot override `onClick` when doing `nextjs` client side routing. You can check this by searching for an app in global search and selecting an option. The whole page will refresh before taking you to the launch page.

Reply:
Good catch - I think I've got a fix in place. And the screenshots have been updated.